src/resolvers/user-resolver.ts

```diff
@@ -17,22 +17,69 @@ const checkEmailExistence = async (email: string): Promise<void> => {
 const createNewUser = async (userData: UserInput): Promise<User> => {
   const hashedPassword = await bcrypt.hash(userData.password, 10);
 
-  return prisma.user.create({
+  return await prisma.user.create({
     data: {
       ...userData,
       password: hashedPassword,
       birthDate: new Date(userData.birthDate),
+      addresses: { create: userData.addresses },
     },
+    include: { addresses: true },
   });
 };
 
 export const resolvers = {
   Query: {
-    users: async (): Promise<User[]> => {
-      return await prisma.user.findMany();
+    user: async (_: unknown, args: { id: string }, context: { user: string | null }): Promise<User> => {
+      const { id } = args;
+
+      validateAuth(context.user);
+
+      const user = await prisma.user.findUnique({
+        where: { id: Number(id) },
+        include: { addresses: true },
+      });
+
+      if (!user) {
+        throw new CustomError('Usuário não encontrado.', 404, 'Tente novamente com um ID válido.');
+      }
+
+      return user;
     },
-    hello: async (): Promise<string> => {
-      return 'Hello, World!';
+
+    Users: async (
+      _: unknown,
+      args: { pageData: PaginationInput },
+      context: { user: string | null },
+    ): Promise<UserPagination> => {
+      const { pageData } = args;
+
+      validateAuth(context.user);
+
+      if (pageData.limit <= 0) {
+        throw new CustomError('Quantidade inválida.', 400, 'A quantidade de usuários deve ser maior que zero.');
+      }
+
+      const users = await prisma.user.findMany({
+        take: pageData.limit,
+        orderBy: { name: 'asc' },
+        skip: pageData.offset,
+        include: { addresses: true },
+      });
+
+      const usersPreviousPage: PaginationInfo = {
+        hasMoreUsers: pageData.offset > 0,
+        totalUsersInPage: pageData.offset,
+      };
+
+      const lastUser = pageData.offset + pageData.limit;
+      const totalUsers = await prisma.user.count();
+      const usersNextPage: PaginationInfo = {
+        hasMoreUsers: lastUser < totalUsers,
+        totalUsersInPage: totalUsers - lastUser > 0 ? totalUsers - lastUser : 0,
```

Comment:
aqui vc faz a mesma validação né, daria pra separar em um const
```suggestion
       const hasMoreUsers = lastUser < totalUsers // 0 < totalUsers - lastUser

        hasMoreUsers,
        totalUsersInPage: hasMoreUsers ? totalUsers - lastUser : 0,
```